runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java

```diff
@@ -219,25 +226,47 @@ public void onLookupMenuClicked(PlayerMenuOptionClicked event)
 	{
 		if (event.getMenuOption().equals(LOOKUP))
 		{
-			executor.execute(() ->
+			lookupPlayer(Text.removeTags(event.getMenuTarget()));
+		}
+	}
+
+	private void lookupPlayer(String playerName)
+	{
+		executor.execute(() ->
+		{
+			try
 			{
-				try
+				SwingUtilities.invokeAndWait(() ->
 				{
-					SwingUtilities.invokeAndWait(() ->
+					if (!navButton.isSelected())
 					{
-						if (!navButton.isSelected())
-						{
-							navButton.getOnSelect().run();
-						}
-					});
-				}
-				catch (InterruptedException | InvocationTargetException e)
+						navButton.getOnSelect().run();
+					}
+				});
+			}
+			catch (InterruptedException | InvocationTargetException e)
+			{
+				throw new RuntimeException(e);
+			}
+
+			hiscorePanel.lookup(playerName);
+		});
+	}
+
+	@Subscribe
+	public void onChatMessageReceived(ChatMessage event)
+	{
+		if (config.bountylookup() && event.getType().equals(ChatMessageType.SERVER))
+		{
+			String message = event.getMessage();
+			if (message.startsWith("<col=ff0000>You've been assigned a target: "))
```

Comment:
Is there a reason this can't use `bountyPattern` to check if it matched?

Reply:
No there is not, I changed it to make use of bountyPattern, testing that now to see if I can find any issues with doing it that way.